src/scripts/main.js

```diff
@@ -1,3 +1,72 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const rows = table.rows;
+
+const appendRow = document.querySelector('.append-row');
+const removeRow = document.querySelector('.remove-row');
+const appendColumn = document.querySelector('.append-column');
+const removeColumn = document.querySelector('.remove-column');
+
+const minCells = 2;
+const maxCells = 10;
+
+appendRow.onclick = function () {
+  if (rows.length < maxCells) {
+    const newRow = table.insertRow();
+
+    const columnCount = rows[0].cells.length;
+
+    for (let i = 0; i < columnCount; i++) {
+      newRow.insertCell(-1);
+    }
+
+    removeRow.disabled = false;
+
+    if (rows.length === maxCells) {
+      appendRow.disabled = true;
+    }
+  }
+};
+
+removeRow.onclick = function () {
+  table.deleteRow(-1);
+
+  appendRow.disabled = false;
+
+  if (rows.length === minCells) {
+    removeRow.disabled = true;
+  }
+};
+
+appendColumn.onclick = function () {
+  const columnCount = rows[0].cells.length;
```

Comment:
Code Quality Issue: The 'columnCount' variable is declared multiple times in different functions. Consider declaring it once in a higher scope to improve code readability and performance.